crmsh/upgradeutil.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import os.path
+import parallax
+import subprocess
+import typing
+
+from . import parallax as crmparallax
+from . import utils as crmutils
+
+
+# pump this seq when upgrade check need to be run
+CURRENT_UPGRADE_SEQ = 1
+DATA_DIR = os.path.expanduser('~hacluster/crmsh')
+SEQ_FILE_PATH = DATA_DIR + '/upgrade_seq'
+FORCE_UPGRADE_FILE_PATH = DATA_DIR + '/upgrade_forced'
+
+
+logger = logging.getLogger(__name__)
+
+
+class _SkipUpgrade(Exception):
+    pass
+
+
+def _get_file_content(path, default=None):
+    try:
+        with open(path, 'rb') as f:
+            return f.read()
+    except FileNotFoundError:
+        return default
+
+
+def _parallax_run(nodes, cmd):
+    parallax_options = parallax.Options()
+    parallax_options.ssh_options = ['StrictHostKeyChecking=no', 'ConnectTimeout=10']
+    return parallax.run(nodes, cmd, parallax_options)
+
+
+def _upgrade_if_needed(nodes, fn: typing.Callable[[int], None]):
+    seq = 0
+    forced = 0 == subprocess.call(['test', '-f', FORCE_UPGRADE_FILE_PATH])
+    needed = forced
```

Comment:
Since `forced` not used again in this function, how about
```
needed = 0 == subprocess.call(['test', '-f', FORCE_UPGRADE_FILE_PATH])
```
directly?